collectors/proc.plugin/sys_class_drm.c

```diff
@@ -0,0 +1,1148 @@
+        rrddim_set_by_pointer(c->st_mem_usage_vis_vram, c->rd_mem_free_vis_vram, c->total_vis_vram - c->used_vis_vram);
+        rrdset_done(c->st_mem_usage_vis_vram);
+        return 0;
+    }
+    else {
+        collector_error("Cannot read used_vis_vram for %s: [%s]", c->pathname, c->id.marketing_name);
+        freez((void *) c->pathname_mem_used_vis_vram);
+        freez((void *) c->pathname_mem_total_vis_vram);
+        rrdset_is_obsolete(c->st_mem_usage_perc_vis_vram);
+        rrdset_is_obsolete(c->st_mem_usage_vis_vram);
+        return 1;
+    }
+}
+
+static int do_rrd_gtt(struct card *const c){
+    if(likely(!read_single_number_file(c->pathname_mem_used_gtt, (unsigned long long *) &c->used_gtt) && 
+            c->total_gtt)){
+        rrddim_set_by_pointer(  c->st_mem_usage_perc_gtt, 
+                                c->rd_mem_used_perc_gtt, 
+                                c->used_gtt * 10000 / c->total_gtt);
+        rrdset_done(c->st_mem_usage_perc_gtt);
+    
+        rrddim_set_by_pointer(c->st_mem_usage_gtt, c->rd_mem_used_gtt, c->used_gtt);
+        rrddim_set_by_pointer(c->st_mem_usage_gtt, c->rd_mem_free_gtt, c->total_gtt - c->used_gtt);
+        rrdset_done(c->st_mem_usage_gtt);
+        return 0;
+    }
+    else {
+        collector_error("Cannot read used_gtt for %s: [%s]", c->pathname, c->id.marketing_name);
+        freez((void *) c->pathname_mem_used_gtt);
+        freez((void *) c->pathname_mem_total_gtt);
+        rrdset_is_obsolete(c->st_mem_usage_perc_gtt);
+        rrdset_is_obsolete(c->st_mem_usage_gtt);
+        return 1;
+    }
+}
+
+static char *set_id(const char *const suf_1, const char *const suf_2, const char *const suf_3){
+    static char id[RRD_ID_LENGTH_MAX + 1];
+    snprintfz(id, RRD_ID_LENGTH_MAX, "%s_%s_%s", suf_1, suf_2, suf_3);
+    return id;
+}
+
+int do_sys_class_drm(int update_every, usec_t dt) {
+    (void)dt;
+    
+    static DIR *drm_dir = NULL;
+
+    int chart_prio = NETDATA_CHART_PRIO_DRM_AMDGPU;
+
+    if(unlikely(!drm_dir)) {
+        char filename[FILENAME_MAX + 1];
+        snprintfz(filename, FILENAME_MAX, "%s%s", netdata_configured_host_prefix, "/sys/class/drm");
+        char *drm_dir_name = config_get(CONFIG_SECTION_PLUGIN_PROC_DRM, "directory to monitor", filename);
+        if(unlikely(NULL == (drm_dir = opendir(drm_dir_name)))){
+            collector_error("Cannot read directory '%s'", drm_dir_name);
+            return 1;
+        }
+
+        struct dirent *de = NULL;
+        while(likely(de = readdir(drm_dir))) {
+            if( de->d_type == DT_DIR && ((de->d_name[0] == '.' && de->d_name[1] == '\0') || 
+                 (de->d_name[0] == '.' && de->d_name[1] == '.' && de->d_name[2] == '\0'))) continue;
+            
+            if(de->d_type == DT_LNK && !strncmp(de->d_name, "card", 4) && !strchr(de->d_name, '-')) {
+                struct card *const c = callocz(1, sizeof(struct card));  
+
+                /* Get static info */              
```

Comment:
All AMD cards have `DRIVER=amdgpu` in `/sys/class/drm/cardX/device/uevent`. I think you need to check this, it will remove log messages like on every (re)start

```
2023-08-07 14:42:05: netdata INFO  : P[proc] : Cannot read file '/sys/class/drm/card0/device/gpu_busy_percent'
2023-08-07 14:42:05: netdata INFO  : P[proc] : Cannot read file '/sys/class/drm/card0/device/mem_busy_percent'
2023-08-07 14:42:05: netdata INFO  : P[proc] : Cannot read file '/sys/class/drm/card0/device/pp_dpm_sclk'
2023-08-07 14:42:05: netdata INFO  : P[proc] : Cannot read file '/sys/class/drm/card0/device/pp_dpm_mclk'
2023-08-07 14:42:05: netdata INFO  : P[proc] : Cannot read file '/sys/class/drm/card0/device/mem_info_vram_used'
2023-08-07 14:42:05: netdata INFO  : P[proc] : Cannot read file '/sys/class/drm/card0/device/mem_info_vram_total'
2023-08-07 14:42:05: netdata INFO  : P[proc] : Cannot read file '/sys/class/drm/card0/device/mem_info_vis_vram_used'
2023-08-07 14:42:05: netdata INFO  : P[proc] : Cannot read file '/sys/class/drm/card0/device/mem_info_vis_vram_total'
2023-08-07 14:42:05: netdata INFO  : P[proc] : Cannot read file '/sys/class/drm/card0/device/mem_info_gtt_used'
```

Reply:
Done